### PR TITLE
Cleanups and optimizations

### PR DESCRIPTION
**Pull request by phlax**
- use_caching in signals
- select_relateds in update code
- use bulk_create for adding qualitychecks on unit.save

### CODE REVIEW

File: pootle/apps/pootle_translationproject/utils.py [hunk 1]
```diff
@@ -156,9 +159,10 @@ def update_children(self, source_dir, target_dir, update_cache=True):
         """
         stores = []
         dirs = []
-        for store in source_dir.child_stores.live():
+        for store in source_dir.child_stores.live().select_related("parent", "translation_project"):
             stores.append(store.name)
             try:
+                # Force validation of fields.
```
Comment (unho) [inline pootle/apps/pootle_translationproject/utils.py]: Is this a TODO note? Or is it stating that the following code enforces validation?

Comment (phlax) [inline pootle/apps/pootle_translationproject/utils.py]: copypasta

File: pootle/apps/pootle_translationproject/utils.py [hunk 2]
```diff
@@ -156,9 +159,10 @@ def update_children(self, source_dir, target_dir, update_cache=True):
         """
         stores = []
         dirs = []
-        for store in source_dir.child_stores.live():
+        for store in source_dir.child_stores.live().select_related("parent", "translation_project"):
```
Comment (ta2-1) [inline pootle/apps/pootle_translationproject/utils.py]: may be `store.parent = source_dir` instead of `parent` in select_related?

**Comment by unho** [comment]
Please add some explanation on the commit message about the undocumented `use_caching`. Other than that lgtm.

**Pull request merged**